src/components/genericFormMaker.js

```diff
@@ -19,7 +19,7 @@ export default function genericFormMaker(info) {
     constructor(props) {
       super(props);
       this.state = {
-        data: props.formData,
+        data: props.update ? props.formData : props.formData.data,
```

Comment:
Why is there a ternary operator used? Why can't we `props.formData.data` for both cases? You nicely explained how the state is being updated for `this.state.data.data` so won't it be better if we rectify this behaviour?

Reply:
That is because `add` and `edit` uses different props. `add`'s `props.formData` will have an object defined in `initial.js`, whereas `edit`'s `props.formData` will have actual details of that particular instance.